pkg/apis/v1alpha1/suite_test.go

```diff
@@ -115,4 +120,93 @@ var _ = Describe("Validation", func() {
+				DetailedMonitoring: aws.Bool(false),
+			}
+			awsnodetemplate = test.AWSNodeTemplate(awsnodetemplatespec)
+		})
+		DescribeTable("should change hash when static fields are updated", func(awsnodetemplatespec v1alpha1.AWSNodeTemplateSpec) {
+			expectedHash := awsnodetemplate.Hash()
+			updatedAWSNodeTemplate := test.AWSNodeTemplate(*awsnodetemplatespec.DeepCopy(), awsnodetemplatespec)
+			actualHash := updatedAWSNodeTemplate.Hash()
+			Expect(actualHash).ToNot(Equal(fmt.Sprint(expectedHash)))
+		},
+			Entry("InstanceProfile Drift", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{InstanceProfile: aws.String("profile-2")}}),
+			Entry("UserData Drift", v1alpha1.AWSNodeTemplateSpec{UserData: aws.String("userdata-test-2")}),
+			Entry("Tags Drift", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{Tags: map[string]string{"keyTag-test-3": "valueTag-test-3"}}}),
+			Entry("MetadataOptions Drift", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{LaunchTemplate: v1alpha1.LaunchTemplate{MetadataOptions: &v1alpha1.MetadataOptions{HTTPEndpoint: aws.String("test-metadata-2")}}}}),
+			Entry("BlockDeviceMappings Drift", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{LaunchTemplate: v1alpha1.LaunchTemplate{BlockDeviceMappings: []*v1alpha1.BlockDeviceMapping{{DeviceName: aws.String("map-device-test-3")}}}}}),
+			Entry("Context Drift", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{Context: aws.String("context-2")}}),
+			Entry("DetailedMonitoring Drift", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: aws.Bool(true)}),
+		)
+		It("should not change hash when behavior/dynamic fields are updated", func() {
+			actualHash := awsnodetemplate.Hash()
+
+			expectedHash, err := hashstructure.Hash(awsnodetemplate.Spec, hashstructure.FormatV2, &hashstructure.HashOptions{
+				SlicesAsSets:    true,
+				IgnoreZeroValue: true,
+				ZeroNil:         true,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actualHash).To(Equal(fmt.Sprint(expectedHash)))
+
+			// Update a behavior/dynamic field
+			awsnodetemplate.Spec.SubnetSelector = map[string]string{"subnet-test-key": "subnet-test-value"}
+			awsnodetemplate.Spec.SecurityGroupSelector = map[string]string{"sg-test-key": "sg-test-value"}
+			awsnodetemplate.Spec.AMIFamily = aws.String("test-family")
+			awsnodetemplate.Spec.AMISelector = map[string]string{"ami-test-key": "ami-test-value"}
+
+			actualHash = awsnodetemplate.Hash()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actualHash).To(Equal(fmt.Sprint(expectedHash)))
+		})
+		It("should expect two provisioner with the same spec to have the same provisioner hash", func() {
+			awsnodetemplateTwo := &v1alpha1.AWSNodeTemplate{
+				ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+			}
+			awsnodetemplateTwo.Spec = awsnodetemplatespec
+
+			Expect(awsnodetemplate.Hash()).To(Equal(awsnodetemplateTwo.Hash()))
+		})
+		It("should expect fields that are reordered to not produce a new hash", func() {
```

Comment:
`DescribeTable` I think here too. I think if you find yourself iterating over what seems like a test case in an `It` block, we should be using a `DescribeTable`